map_model/src/objects/building.rs

```diff
@@ -346,17 +429,26 @@ impl AmenityType {
 
     pub fn parse(x: &str) -> Option<AmenityType> {
```

Comment:
It feels like we're outgrowing manually writing these functions to go to/from a string. Glancing around, https://docs.rs/strum/0.20.0/strum/derive.EnumString.html is a popular crate for auto-deriving this with macros. I'm generally hesitant to pile on more dependencies and have the build time/size creep up, but it looks like we're already pulling in the heavier proc macro crates, so I think it'd be worth trying this. The `strum` docs are a little hard to follow, and the crate does lots of stuff we don't seem to need, but https://github.com/Peternator7/strum/blob/master/strum_tests/tests/from_str.rs seems simple to start with

Reply:
Totally agree, as I was ordering them and copying things around it was quite a pain to keep repeating things. I will take a stab with strum.